src/CareTogether.Core/Engines/PolicyEvaluation/ReferralCalculations.cs

```diff
@@ -528,29 +519,38 @@ ImmutableList<DateOnly> completionDates
 
         private static IEnumerable<DateRange<Guid>> GenerateDateRanges(ImmutableList<ChildLocation> childLocations)
         {
-            DateOnly? startDate = null;
-            Guid? tag = null;
+            (DateOnly, Guid)? entry = null;
```

Comment:
Is it possible to use an anonymous record here?

Reply:
Try `(DateOnly myDate, Guid myId)?`

Reply:
Rename to `previousNotPausedEntry` for clarity